docs/transactions/source/construct-a-tx.rst

```diff
@@ -23,41 +23,41 @@ This page lists the steps to construct a valid transaction.
 #. Construct an :term:`associative array` named ``unfulfilled_tx`` of the form
    (based on JSON syntax):
 
-.. code-block:: bash
+    .. code-block:: bash
 
-   {
-      "id": null,
-      "version": version,
-      "inputs": unfulfilled_inputs,
-      "outputs": outputs,
-      "operation": operation,
-      "asset": asset,
-      "metadata": metadata
-    }
+      {
+          "id": null,
+          "version": version,
+          "inputs": unfulfilled_inputs,
+          "outputs": outputs,
+          "operation": operation,
+          "asset": asset,
+          "metadata": metadata
+        }
 
-Note how ``unfulfilled_tx`` includes a key-value pair for the ``"id"`` key.
-The value must be your programming language's equivalent of :term:`null`.
-
-We will now convert ``unfulfilled_tx`` to something
-that can be signed:
+    Note how ``unfulfilled_tx`` includes a key-value pair for the ``"id"`` key.
+    The value must be your programming language's equivalent of :term:`null`.
 
 #. :ref:`Convert unfulfilled_tx to an IPDB-standard JSON string
    <JSON Serialization & Deserialization>` named ``utx_json``.
-#. :ref:`Convert utx_json to bytes <Converting Strings to Bytes>`.
-   Call the result ``utx_bytes``.
-
-``utx_bytes`` can be signed to compute all signatures and fulfillment strings.
-
 #. Create ``inputs`` as a deep copy of ``unfulfilled_inputs``.
-#. For each input in ``inputs``,
-   fulfill the associated crypto-condition
-   `using an implementation of crypto-conditions
-   <https://github.com/rfcs/crypto-conditions#implementations>`_.
-   You will need ``utx_bytes`` and one or more private keys
-   (which are used to sign ``utx_bytes``).
-   The end result is usually some kind of fulfilled condition object.
-   Compute the fulfillment string of that fulfilled condition object, and
-   put that as the value of ``"fulfillment"`` for the input in question.
+#. For each input in ``inputs``:
+
+    #. If ``operation`` is not ``"TRANSFER"``,
+       let ``string_to_sign = utx_json``,
+       otherwise let ``string_to_sign = utx_json + output_tx_id + output_index``
+       where ``output_tx_id`` is the transaction ID of the output that
+       this input fulfills and ``+`` means concatenate the strings.
```

Comment:
This does not need to be dependent on the operation type. As long as there is an input pointing to an output, the message to sign needs to be the hash of the concatenation of the transaction, utxo tx id and utxo output index.

Reply:
I see, that makes sense. I'll change it.